packages/embark-deploy-tracker/src/index.js

```diff
@@ -37,8 +37,10 @@ class DeployTracker {
     this.embark.registerActionForEvent("deploy:beforeAll", this.setCurrentChain.bind(this));
 
     this.events.on("deploy:contract:deployed", (contract) => {
-      self.trackContract(contract.className, contract.realRuntimeBytecode, contract.realArgs, contract.deployedAddress);
-      self.save();
+      if (contract.realArgs !== undefined) {
+        self.trackContract(contract.className, contract.realRuntimeBytecode, contract.realArgs, contract.deployedAddress);
+        self.save();
+      }
```

Comment:
Alright this is kind of hacky but I had decide between two options...

Basically we're dealing with a new case here. If a contract isn't tracked yet but signalled as "deployed" Embark will try to track it. Obviously it doesn't really make sense to track contracts that, per definition, are already deployed somewhere at some point for which reason we have provided an `address`.

However, deploy-tracker doesn't care. It'll try to track everything that's signalled as `deployed` as it simply reacts to `deploy:contract:deploy` event. It seems however that, when `address === function` the different code path causes `contract.realArgs` to be `undefined`. In a way it makes sense because there's no point in defining args when defining an `address`. Not sure why this isn't a problem when `address === 'string'`. 

Either way, this will break Embark at this point, so I decided to check for realArgs being existent here. 

I'm aware this is a horrible hack, so I'd propose to do the following:

- introduce a new command handler `tracker:track:contract` inside deploy-tracker
- have components explicitly ask for tracking a contract (instead of it being an implicit side effect)
- remove trackers listener to `deploy:contract:deployed`. It shouldn't care it about it, instead it should do it's work when somebody asks about it.

That way we introduce a new API that allows us to control when tracking should happen (e.g. in the case of `address == function | string` we don't want to track).

Thoughts?

Reply:
Another thought I wanted to add: You might wonder: why are we emitting `deploy:contract:deployed` in the first place when we know the contract is already deployed?

Valid question. The reason for that is that we need to update the contract status in the dashboard. This one only allows for 2-3 events to update the state. `deploy:contract:deployed` being one of them.

Reply:
Good find and good suggestions. I'm not sure what's the best solution right now, but your hack is fine for the moment.
If you would just add a comment with a TODO for a refactor or something like that, it would be great.